front/app/components/UI/UserSelect/index.tsx

```diff
@@ -0,0 +1,141 @@
+  users: GetUsersChildProps;
+}
+
+interface Props {
+  onChange: (id: string) => void;
+  value: string;
+  placeholder?: string;
+  disabled?: boolean;
+  className?: string;
+  id: string;
+  inputId: string;
+}
+
+export const AvatarImage = styled.img`
+  flex: 0 0 30px;
+  width: 30px;
+  height: 30px;
+  fill: #596b7a;
+  padding: 15px;
+  border-radius: 50%;
+  background: white;
+  margin-right: 0.5rem;
+`;
+
+const AvatarIcon = styled(Icon)`
+  flex: 0 0 30px;
+  width: 30px;
+  height: 30px;
+  background: white;
+  border-radius: 50%;
+  fill: #596b7a;
+  margin-right: 0.5rem;
+`;
+
+const UserOption = styled.div`
+  display: flex;
+  align-items: center;
+`;
+
+const UserSelect = ({
+  users,
+  onChange,
+  value,
+  placeholder,
+  disabled = false,
+  className,
+  id,
+  inputId,
+}: DataProps & Props): ReactElement => {
+  const usersList = Array.isArray(users.usersList) ? users.usersList : [];
+
+  const handleChange = (option: OptionTypeBase) => {
+    onChange(option.id);
+  };
+
+  const filterByNameAndEmail = (option: OptionTypeBase, searchText: string) => {
+    if (
+      option.data.attributes.first_name
+        .toLowerCase()
+        .includes(searchText.toLowerCase()) ||
+      option.data.attributes.last_name
+        .toLowerCase()
+        .includes(searchText.toLowerCase()) ||
+      option.data.attributes.email
+        .toLowerCase()
+        .includes(searchText.toLowerCase())
+    ) {
+      return true;
+    } else {
+      return false;
+    }
+  };
+
+  const selectedUser = usersList.find((user) => user.id === value);
+
+  const Avatar = ({ userId }) => {
+    const user = usersList.find((user) => user.id === userId);
+    const avatarSrc =
+      user?.attributes?.avatar?.medium || user?.attributes?.avatar?.small;
+    return (
+      <>
+        {avatarSrc ? (
+          <AvatarImage className="avatarImage" src={avatarSrc} alt="" />
+        ) : (
+          <AvatarIcon className="avatarIcon" name="user" />
+        )}
+      </>
+    );
+  };
+
+  const getOptionLabel = (option: OptionTypeBase): any => (
+    <UserOption>
+      <Avatar userId={option.value} />
+      {option.attributes.first_name} {option.attributes.last_name} (
+      {option.attributes.email})
+    </UserOption>
+  );
+
+  const getOptionId = (option: OptionTypeBase) => option.id;
+
+  return (
+    <ReactSelect
+      id={id}
+      inputId={inputId}
+      className={className}
+      isSearchable
+      blurInputOnSelect
+      backspaceRemovesValue={false}
+      menuShouldScrollIntoView={false}
+      isClearable={false}
+      value={selectedUser}
+      placeholder={placeholder as string}
+      options={usersList}
+      getOptionValue={getOptionId}
+      getOptionLabel={getOptionLabel}
+      onChange={handleChange}
+      isDisabled={disabled}
+      menuPlacement="auto"
+      styles={selectStyles}
+      filterOption={filterByNameAndEmail}
+    />
+  );
+};
+
+const Data = adopt<DataProps>({
+  users: <GetUsers />,
```

Comment:
If I'm not mistaken GetUsers will by default be paginated to only retrieve 20 users. Not sure if that's a problem here?

Reply:
oh yes thats a problem, not paginating on the other hand is probably an issue too.. I'll probably have to change stuff to perform the search directly in the BE

Reply:
Yes indeed, setting the page size to a super high value migth indeed not be the best solution :). GetUsers already has a search prop so that migth be worth a shot.

Reply:
I did that and added a load more function too, could you give it a second look?

Reply:
Looks good to me!